packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -0,0 +1,182 @@
+  ///
+  /// `capture` is only supported in mobile browsers.
+  /// See https://caniuse.com/#feat=html-media-capture
+  @visibleForTesting
+  Future<PickedFile> pickFile({
+    String accept,
+    String capture,
+  }) {
+    html.FileUploadInputElement input = createInputElement(accept, capture);
+    _injectAndActivate(input);
+    return _getSelectedFile(input);
+  }
+
+  // DOM methods
+
+  /// Converts plugin configuration into a proper value for the `capture` attribute.
+  ///
+  /// See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#capture
+  @visibleForTesting
+  String computeCaptureAttribute(ImageSource source, CameraDevice device) {
+    String capture;
+    if (source == ImageSource.camera) {
+      capture = device == CameraDevice.front ? 'user' : 'environment';
+    }
+    return capture;
+  }
+
+  html.File _getFileFromInput(html.FileUploadInputElement input) {
+    if (_hasOverrides) {
+      return _overrides.getFileFromInput(input);
+    }
+    return input.files[0];
+  }
+
+  /// Handles the OnChange event from a FileUploadInputElement object
+  /// Returns the objectURL of the selected file.
+  String _handleOnChangeEvent(html.Event event) {
+    // load the file...
+    final html.FileUploadInputElement input = event.target;
+    final html.File file = _getFileFromInput(input);
+
+    if (file != null) {
+      return html.Url.createObjectUrl(file);
+    }
+    return null;
+  }
+
+  /// Monitors an <input type="file"> and returns the selected file.
+  Future<PickedFile> _getSelectedFile(html.FileUploadInputElement input) async {
+    // Observe the input until we can return something
+    final Completer<PickedFile> _completer = Completer<PickedFile>();
+    input.onChange.listen((html.Event event) async {
+      final objectUrl = _handleOnChangeEvent(event);
+      _completer.complete(PickedFile(objectUrl));
+    });
+    input.onError // What other events signal failure?
+        .listen((html.Event event) {
+      _completer.completeError(event);
+    });
+
+    return _completer.future;
+  }
+
+  /// Initializes a DOM container where we can host input elements.
+  html.Element _initTarget(String id) {
+    var target = html.querySelector('#${id}');
+    if (target == null) {
+      final html.Element targetElement =
+          html.Element.tag('flt-image-picker-inputs')..id = id;
+
+      html.querySelector('body').children.add(targetElement);
+      target = targetElement;
+    }
+    return target;
+  }
+
+  /// Creates an input element that accepts certain file types, and
+  /// allows to `capture` from the device's cameras (where supported)
+  @visibleForTesting
+  html.Element createInputElement(String accept, String capture) {
+    if (_hasOverrides) {
+      return _overrides.createInputElement(accept, capture);
+    }
+
+    html.Element element;
+
+    if (capture != null) {
+      // Capture is not supported by dart:html :/
+      element = html.Element.html(
```

Comment:
I'd use JS-interop to set the capture attribute. Passing node validators results in unnecessary security reviews later on.

Reply:
That's 2 writes to the DOM, instead of 1, but I'll give this a shot (since I have the html.Element, I should be able to set the capture attribute easily).

Thanks for the insight on the extra security reviews, I don't want to be involved in those unless they're really warranted :P

Reply:
Ah no, this ended up being mmmmuch cleaner than it was before:

```dart
    html.Element element = html.FileUploadInputElement()..accept = accept;
    
    if (capture != null) {
      element.setAttribute('capture', capture);
    }
```